### PR TITLE
feat: Add condition list to extensibility and add status section

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- add status section in extensibility
- implement condition list for extensibility
- document ConditionList

**Related issue(s)**
https://github.com/kyma-project/busola/issues/2886

### CODE REVIEW

File: src/shared/components/ResourceDetails/ResourceDetails.js [hunk 1]
```diff
@@ -281,15 +282,43 @@ function Resource({
...
+                    style={spacing.sapUiSmallMarginBeginEnd}
+                  >
+                    {component.header}:
+                  </div>
+                  {component.value(resource)}
+                </>
+              ))}
+          </>
+        ) : null
+      }
```
Comment (OliwiaGowor) [inline src/shared/components/ResourceDetails/ResourceDetails.js:321]: extract this filtering logic to the ExtensibilityDetails, so we don't have extensibility related logic in the component

**Review by OliwiaGowor — COMMENTED** [review]
- [ ] if there is no info the empty status card is still displayed (for example in Functions, HPA) - it shouldn't be visible
Screenshot 2024-06-04 at 10 03 57

File: docs/extensibility/display-section.md [hunk 2]
```diff
@@ -516,6 +516,20 @@ Columns widgets render the child widgets in multiple columns.
 
 <img src="./assets/display-widgets/Columns.png" alt="Example of a columns widget" style="border: 1px solid #D2D5D9">
 
+### ConditionList
+
+Condition List widget renders the conditions as expandable list with condition details.
```
Comment (grego952) [inline docs/extensibility/display-section.md:521]:
```suggestion
The condition List widget renders the conditions as an expandable list with condition details.
```

**Review by grego952 — APPROVED** [review]

File: src/components/Extensibility/ExtensibilityDetails.js [hunk 3]
```diff
@@ -119,6 +120,44 @@ export const ExtensibilityDetailsCore = ({
             ]
           : []
       }
+      customStatusColumns={
```
Comment (akucharska) [inline src/components/Extensibility/ExtensibilityDetails.js:123]: We still render empty div even if `customStatusColumns` are empty
Screenshot 2024-06-07 at 09 20 13

File: docs/extensibility/30-details-summary.md [hunk 4]
```diff
@@ -4,20 +4,20 @@ You can customize the details page of the user interface component of your resou
 
 ## Available Parameters
 
-In the **data.details** section you can provide configuration of three optional components: **header**, **body**, and **resourceGraph**. The **header** and **body** components are lists of widgets visible in the respective sections of the details page. You can use the **resourceGraph** component to present the relationship between different resources.
+In the **data.details** section you can provide configuration of four optional components: **header**, **body**, **status** and **resourceGraph**. The **header**, **status** and **body** components are lists of widgets visible in the respective sections of the details page. You can use the **resourceGraph** component to present the relationship between different resources.
 
 ### **header** and **body** Parameters
```
Comment (akucharska) [inline docs/extensibility/30-details-summary.md:9]:
```suggestion
### **header** , **status**  and **body** Parameters
```

**Review by akucharska — APPROVED** [review]

**Pull request merged**